src/retractions.jl

```diff
@@ -27,6 +27,23 @@ An abstract type for representing approximate retraction methods.
 """
 abstract type ApproximateRetraction <: AbstractRetractionMethod end
 
+
+"""
+    ApproximateExponentialRetraction{N <: NamedTuple} <: ApproximateRetraction
+
+Retraction using an algorithm with certain parameters to approximate the exponential map.
+
+# Fields
+* `parameters` – a named tuple containing the parameters for the approximation algorithm
+
+# Constructor
+
+    ApproximateExponentialRetraction(parameters::NamedTuple)
+"""
+struct ApproximateExponentialRetraction{N <: NamedTuple} <: ApproximateRetraction
+    parameters::N
```

Comment:
I'd suggest renaming it to `keyword_arguments` or `kwargs` for short to not confuse people that they might be positional.

Reply:
sure, can do that tomorrow.